web/src/pages/register.tsx

```diff
@@ -0,0 +1,69 @@
+import Link from 'next/link';
+import { useState, ChangeEvent } from 'react';
+import { MdArrowBackIosNew } from 'react-icons/md';
+import Button from '../components/Inputs/Button';
+import LinkIconButton from '../components/Inputs/LinkIconButton';
+import TextField from '../components/Inputs/TextField';
+
+
+export default function RegisterPage() {
+
+    const [state, setState] = useState({
+        email:"",
+        username: "",
+        password: "",
+        retype: ""
+    });
+
+    const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+        setState(prev => ({...prev, [e.target.name]:e.target.value}));
+    }
```

Comment:
This is clever, but unidiomatic. I would favor just using 4 `useState()` statements